### PR TITLE
[FrameworkBundle] Document env vars in framework.ide

### LINKED ISSUE #15303: [FrameworkBundle][HttpKernel] Move IDE file link formats from Framework…
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#41002
| PR author(s) | @MatTheCat
| Merged in    | 5.3-dev

### PR DESCRIPTION
**Pull request by javiereguiluz**
This fixes #15303.

@MatTheCat, please review this. Thank you!

### CODE REVIEW

File: reference/configuration/framework.rst [hunk 1]
```diff
@@ -461,10 +461,52 @@ doubling them to prevent Symfony from interpreting them as container parameters)
...
+        use Symfony\Config\FrameworkConfig;
+
+        return static function (FrameworkConfig $framework) {
+            // the env var stores the IDE/editor name (e.g. 'phpstorm', 'vscode', etc.)
+            $framework->ide('%env(resolve:CODE_EDITOR)%');
+        };
+
+.. versionadded:: 5.3
+
+    The option to use env vars in the ``frameowrk.ide`` option was introduced
```
Comment (noniagriconomie) [inline reference/configuration/framework.rst]: typo
```suggestion
    The option to use env vars in the ``framework.ide`` option was introduced
```

Comment (javiereguiluz) [reply reference/configuration/framework.rst]: Fixed! Thanks!

**Comment by javiereguiluz** [comment]
@Nyholm there's a validation error related to a code example of this article and I don't know how to fix it. If you have some time, please give me a clue to solve it. Thanks!

**Comment by Nyholm** [comment]
Thank you for the ping. 

The error message basically means: "if you copy/paste this code example to your app, you would get the following error message: "

> Environment variable not found: "CODE_EDITOR".

To fix this, we could define the environment variable `CODE_EDITOR` (or ugly trick it by defining parameter `env(CODE_EDITOR): ''`. 

In this particular case (considering how we usually do examples) I suggest just to ignore this error.

**Comment by javiereguiluz** [comment]
Thanks for the info ... but sadly we can't ignore this error because it will make all future builds in 5.x and up to fail (and also, GitHub sends an email for each failure, so it will be a bit annoying).

Maybe we can tweak the validator to never check if an env var exists?

**Comment by Nyholm** [comment]
Im not sure that is true. 

This CI is using a baseline. So it first detects from where you branched out. It will create the baseline there. Secondly, it applies the patch and run the check again. The errors you see in the PR review is the "new" errors. 

This is the exact workflow symfony/symfony is using for psalm.

**Comment by javiereguiluz** [comment]
OK. I didn't understand the behavior. Sorry! Everything is OK then. Thanks!

File: reference/configuration/framework.rst [hunk 2]
```diff
@@ -461,10 +461,52 @@ doubling them to prevent Symfony from interpreting them as container parameters)
...
+
+        return static function (FrameworkConfig $framework) {
+            // the env var stores the IDE/editor name (e.g. 'phpstorm', 'vscode', etc.)
+            $framework->ide('%env(resolve:CODE_EDITOR)%');
+        };
+
+.. versionadded:: 5.3
+
+    The option to use env vars in the ``framework.ide`` option was introduced
+    in Symfony 5.3.
```
Comment (MatTheCat) [inline reference/configuration/framework.rst:504]: Technically you always could use an env var as `framework.ide` but you wouldn’t get the expected result if it contains an IDE name. I don’t know if we want to mention it though.

Comment (javiereguiluz) [reply reference/configuration/framework.rst:504]: Thanks for the insights ... I'd say it's better to consider that "an internal detail" and not mention it to make docs simpler to read. Thanks!

**Pull request merged**